python/ray/dataframe/concat.py

```diff
@@ -0,0 +1,88 @@
+import pandas as pd
+import numpy as np
+from ray import dataframe as rdf
+from .utils import (
+    from_pandas,
+    _deploy_func)
+from functools import reduce
+
+
+def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
+           keys=None, levels=None, names=None, verify_integrity=False,
+           copy=True):
+
+    def _concat(frame1, frame2):
+        # Check type on objects
+        # Case 1: Both are Pandas DF
+        if isinstance(frame1, pd.DataFrame) and \
+           isinstance(frame2, pd.DataFrame):
+
+            return pd.concat((frame1, frame2), axis, join, join_axes,
+                             ignore_index, keys, levels, names,
+                             verify_integrity, copy)
+
+        # Case 2: Both are different types
+        if isinstance(frame1, pd.DataFrame):
+            frame1 = from_pandas(frame1, len(frame1) / 2**16 + 1)
+        if isinstance(frame2, pd.DataFrame):
+            frame2 = from_pandas(frame2, len(frame2) / 2**16 + 1)
+
+        # Case 3: Both are Ray DF
+        if isinstance(frame1, rdf.DataFrame) and \
+           isinstance(frame2, rdf.DataFrame):
+
+            new_columns = frame1.columns.join(frame2.columns, how=join)
+
+            def remove_columns(pdf):
+                return pdf[new_columns]
+
+            def add_columns(pdf):
+                print("TYPE:", type(pdf))
+                return pdf.reindex(columns=new_columns)
+
+            if axis in [0, 'index', 'rows']:
```

Comment:
What I meant previously was to use the way that Pandas does the error checking. You can call something similar to this line and get the error checking included.

https://github.com/pandas-dev/pandas/blob/v0.22.0/pandas/core/reshape/concat.py#L308

Reply:
Resolved, called said line